pkg/controllers/vdb/init_db.go

```diff
@@ -656,17 +600,13 @@ func (g *GenericDatabaseInitializer) hasCompatibleVersion(supportedVersion, even
 	return ctrl.Result{Requeue: true}
 }
 
-// genCommunalParmsMap takes a multiline string containing the parms already
-// set(in "parm = value" format) and returns the corresponding map.
-func genCommunalParmsMap(content string) map[string]string {
+// genLowerCaseKeysMap will generate a lowercase keys copy of
+// ConfigurationParams map. This is useful to detect duplication
+// between user-defined and operator-generated params.
+func (g *GenericDatabaseInitializer) genLowerCaseKeysMap() map[string]string {
```

Comment:
Could we just ensure the entries into the map always use a lowercase key? Regenerating the map with lowercase keys seems like wasted effort when we can just build the map with lowercase keys to begin with. Here is a suggestion on how to create a case insensitive map: https://stackoverflow.com/questions/11124705/how-to-create-a-case-insensitive-map-in-go